app/js/models/addon_generator.js

```diff
@@ -147,50 +148,107 @@ el.${name} = ${JSON.stringify(value)};
     }
   }
 
-  opMoveAppend(target) {
+  opCopyAppend(destination) {
+    this.operations.push(
+`/*=AddonGenerator::copyAppend*/
+var destination = document.querySelector('${getSelector(destination)}');
+var template = document.createElement('template');
+template.innerHTML = \`${this.target.outerHTML.replace(/\`/g, '\\\`')}\`;
+if (destination) {
+  destination.appendChild(template.content);
+}
+/*==*/`
+    );
+  }
+
+  opCopyPrepend(destination) {
+    this.operations.push(
+`/*=AddonGenerator::copyPrepend*/
+var destination = document.querySelector('${getSelector(destination)}');
+var template = document.createElement('template');
+template.innerHTML = \`${this.target.outerHTML.replace(/\`/g, '\\\`')}\`;
+if (destination) {
+  destination.insertBefore(template.content, destination.firstChild);
+}
+/*==*/`
+    );
+  }
+
+  opCopyAfter(destination) {
+    this.operations.push(
+`/*=AddonGenerator::copyAfter*/
+var destination = document.querySelector('${getSelector(destination)}');
+var template = document.createElement('template');
+template.innerHTML = \`${this.target.outerHTML.replace(/\`/g, '\\\`')}\`;
+if (destination && destination.parentNode) {
+  if (destination.parentNode.lastChild === destination) {
+    destination.parentNode.appendChild(template.content);
+  }
+  else {
+    destination.parentNode.insertBefore(template.content, destination.nextSibling);
+  }
+}
+/*==*/`
+    );
+  }
+
+  opCopyBefore(destination) {
+    this.operations.push(
+`/*=AddonGenerator::copyBefore*/
+var destination = document.querySelector('${getSelector(destination)}');
+var template = document.createElement('template');
+template.innerHTML = \`${this.target.outerHTML.replace(/\`/g, '\\\`')}\`;
+if (destination && destination.parentNode) {
+  destination.parentNode.insertBefore(template.content, destination);
+}
+/*==*/`
+    );
+  }
+
+  opMoveAppend(destination) {
     this.operations.push(
 `/*=AddonGenerator::moveAppend*/
-var target = document.querySelector('${getSelector(target)}');
-if (target) {
-  target.appendChild(el);
+var destination = document.querySelector('${getSelector(destination)}');
+if (destination) {
+  destination.appendChild(el);
 }
 /*==*/`
     );
   }
 
-  opMovePrepend(target) {
+  opMovePrepend(destination) {
     this.operations.push(
 `/*=AddonGenerator::movePrepend*/
-var target = document.querySelector('${getSelector(target)}');
-if (target) {
-  target.insertBefore(el, target.firstChild);
+var destination = document.querySelector('${getSelector(destination)}');
+if (destination) {
+  destination.insertBefore(el, destination.firstChild);
 }
 /*==*/`
     );
   }
 
-  opMoveAfter(target) {
+  opMoveAfter(destination) {
```

Comment:
We could refactor ```opMove***``` and ```opCopy***``` to call into common functions that just do different things with the target.